astropy/utils/data.py

```diff
@@ -997,12 +997,26 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
 
     url_key = remote_url
 
+    # Check if URL is Astropy data server, which has alias, and cache it.
+    if (url_key.startswith(conf.dataurl) and
+            conf.dataurl not in _dataurls_to_alias):
+        with urllib.request.urlopen(conf.dataurl, timeout=timeout) as remote:
```

Comment:
NOTE: Use `contextlib` for PY2 compat (see example in 2.x code just a few lines below this one).